src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -138,11 +138,61 @@ private static bool ApplyChangesFromTC32(JObject sarifLog)
                     RemoveToolLanguage(run);
                     ConvertAllReportingDescriptorNamesToString(run);
                     ConvertAllExceptionMessagesToStringAndRenameToolNotificationNodes(run);
+
+                    // https://github.com/oasis-tcs/sarif-spec/issues/336
+                    UpdateAllToolComponentProperties(run);
                 }
             }
             return true;
         }
 
+        private static void UpdateAllToolComponentProperties(JObject run)
+        {
+            // Access and modify run.tool
+            if (run["tool"] is JObject tool)
+            {
+                UpdateToolObjectToolComponentProperties(tool);
+            }
```

Comment:
>} [](start = 12, length = 1)

we talked about the unsafe cast in the context of a different issue (#1312), so i have added a condition check here. If tool doesnt exist, we can add throwing an error here to fail faster, i wasnt sure if i should just go ahead and do it hence leaving a comment for now :)